add Table of Contents and update API on README.md

mistakia opened the pull request:
Update README with documentation for missing constructor (and options), public methods, and events. In addition, corrects a few event params and uniformly uses `address` to reference the OrbitDB address (instead of inconsistent usage of both `dbname` & `address`).

**Edit**: I have removed two `defaultOptions` properties (`path` & `replicate`) that don't appear to be used. In addition, I have removed the `onProgressCallback` param passed to `loadFromSnapshot()` that was never used.

I've also noticed that a couple events emit inconsistent params.

- `load` : 
https://github.com/orbitdb/orbit-db-store/blob/84aa8095c0d08fc428b8c375aa83cb7fe873bc2d/src/Store.js#L207
https://github.com/orbitdb/orbit-db-store/blob/84aa8095c0d08fc428b8c375aa83cb7fe873bc2d/src/Store.js#L313
- `replicated` : 
https://github.com/orbitdb/orbit-db-store/blob/84aa8095c0d08fc428b8c375aa83cb7fe873bc2d/src/Store.js#L116
https://github.com/orbitdb/orbit-db-store/blob/84aa8095c0d08fc428b8c375aa83cb7fe873bc2d/src/Store.js#L403


Is this desired? or should I update to have matching params?

#### TODO
- [x] document constructor options
- [ ] loadMoreFrom description
- [x] replicated event description
- [x] closed event description
- [x] ready event description
- [x] load event description 

**New scripts**:
 - `npm run build:docs/tocs` : generates a table of contents for README.md

**New dev dependencies**:
- `markdown-toc`: generates a markdown table of contents 

Fixes #26

mistakia commented:
Also, `loadMoreFrom` takes an `amount` param that it never uses:

https://github.com/orbitdb/orbit-db-store/blob/e782db51e9f4bcfb808466fb69bf5169a14308f7/src/Store.js#L272-L274

mistakia commented:
I've noticed that the [main orbit-db Store API section](https://github.com/orbitdb/orbit-db/blob/master/API.md#store-api) is incomplete/needs updating as well.

As mentioned in orbitdb/orbit-db-docstore#10, perhaps I should move this PR upstream to the orbit-db repo and simply link to the relevant API section from this repo's README.md. Either way, for ease of maintenance, it would be best if one repo references the other.

RichardLitt commented on README.md:
```diff
@@ -15,65 +15,113 @@ Base class for [orbit-db](https://github.com/orbitdb/orbit-db) data stores. You
 
 - Node.js >= 8.0.0
 
-### API
```
I would keep the API heading, as a second-level heading.

mistakia replied:
nice catch 👍 updated.

RichardLitt replied:
Could you... add a TOC? :D

mistakia replied:
Could I... borrow your TOC generator?

RichardLitt commented on src/Store.js:
```diff
@@ -309,8 +307,8 @@ class Store {
-  async loadFromSnapshot (onProgressCallback) {
-    this.events.emit('load', this.address.toString())
+  async loadFromSnapshot () {
+    this.events.emit('load', this.address.toString()) //TODO: inconsistent params, line 207
```
Can we fix the params?

mistakia replied:
Not sure what direction to go with on the `load` and `replicated` events. In other words, should the `'load'` event emit the `address` and `heads` or just the `address`.

And with the `'replicated'` event, should it emit the `address` and number of entries replicated? or just the `address`.

RichardLitt replied:
@haadcode should know.

haadcode commented:
It'd be really great to get this PR merged as there is a lot of great documentation here that would help many users.

@mistakia are you interested to continue and finish this up? Totally fine if not and someone else can take a look.

@RichardLitt what do we need to do here from your perspective to be able to merge?

RichardLitt commented:
This looks good to me - I would just work on the params, as adding comments noting it doesn't actually fix the problem. I think @mistakia made a point about what he needs clarified above in [this comment](https://github.com/orbitdb/orbit-db-store/pull/30#discussion_r215607827).

mistakia commented:
Sorry for the delay, just getting back into the swing of things.

@haadcode : As @RichardLitt has pointed out, I just need some clarity on the [load/replicated events](https://github.com/orbitdb/orbit-db-store/pull/30#discussion_r215607827) and I should be able to push the rest over the finish line.

RichardLitt commented:
@aphelionz Would you be able to help here?

aphelionz commented:
@mistakia Is this explanation from the orbit-db pinned issues helpful? https://github.com/orbitdb/orbit-db/issues/579

mistakia commented:
This is ready for review. I updated the docs to make a note that certain arguments are not emitted for `load` and `replicated` events when emitted as a result of `loadFromSnapshot`. Outside of the updates to the documentation, I made a few trivial changes:

- removed an unused argument passed to [loadFromSnapshot](https://github.com/orbitdb/orbit-db-store/pull/30/files#diff-7a95c660669a260e384552aaa39be4bfL313)
- removed an unused property of [DefaultOptions](https://github.com/orbitdb/orbit-db-store/pull/30/files#diff-7a95c660669a260e384552aaa39be4bfL313)

aphelionz commented on README.md:
```diff
@@ -84,56 +160,80 @@ console.log(db.replicationStatus)
+### Private methods
 
-##### `_addOperation(data)`
+#### _addOperation(data)
```
```suggestion
#### `_addOperation(data)`
```

aphelionz commented:
We probably need to update the README a bit now that things like `DefaultOptions` have changed, and there may be some additions/removals to the API in general as well
